mango-core/src/main/java/org/calrissian/mango/domain/event/BaseEvent.java

```diff
@@ -111,8 +91,21 @@ public String toString() {
         return "BaseEvent{" +
                 "type='" + getType() + '\'' +
                 ", id='" + getId() + '\'' +
-                ", timestamp=" + timestamp +
+                ", timestamp=" + getTimestamp() +
                 ", attributes=" + getAttributes() +
                 '}';
     }
+
+    private Predicate<Attribute> filterTimestamp = new Predicate<Attribute>() {
```

Comment:
make this static final so that we don't need to create a new instance per attribute.